server/plugin/webhook.go

```diff
@@ -88,6 +88,9 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if config.EnableWebhookEventLogging {
+		p.API.LogInfo("Webhook Event Log", "event", event)
```

Comment:
The logged data is a bit sparse. Does a methods like https://pkg.go.dev/net/http/httputil#DumpRequest should help here to extract more data?

Reply:
@hanzei checked this package it can help for Http request and response ,
not sure but we can convert the event to string and add the same to log so ll be more easy to read the log

Reply:
I don't quite understand why you want to log the event. Logging the whole request make debugging much easier as we can log it without parsing the payload.

Reply:
I totally misunderstood , i thought of only adding the request body what we get from webhook event to be get logged

we can use this to get the complete request and  can check the logs like 

```
{"timestamp":"2021-10-05 16:19:57.464 +05:30","level":"debug","msg":"Webhook Event Log","caller":"app/plugin_api.go:892","plugin_id":"github","":"event","":"POST /plugins/github/webhook HTTP/1.1\r\nHost: 16da-157-41-140-25.ngrok.io\r\nAccept: */*\r\nAccept-Encoding: gzip\r\nContent-Length: 8654\r\nContent-Type: application/json\r\nUser-Agent: GitHub-Hookshot/24bccdc\r\nX-Forwarded-For: 140.82.115.147\r\nX-Forwarded-Proto: https\r\nX-Github-Delivery: fc320bb6-25c9-11ec-8ffd-dae2201eff9e\r\nX-Github-Event: push\r\nX-Github-Hook-Id: 320917288\r\nX-Github-Hook-Installation-Target-Id: 381551690\r\nX-Github-Hook-Installation-Target-Type: repository\r\nX-Hub-Signature: sha1=e82bc3fbda72fbe834cfcc0025e9ed4902611e0b\r\nX-Hub-Signature-256: sha256=44f69ccbc00d483a9b3b356b32d086e5a7fde280cd82b7c3bc53e5f7d35ba1ba\r\n\r\n{\"ref\":\"refs/heads/sibasankarnayak-patch-2\",\"before\":\"0000000000000000000000000000000000000000\",\"after\":\"3c642e9e366a559f14374fa250ec8815151e7b0f\",\"repository\":{\"id\":381551690,\"node_id\":\"MDEwOlJlcG9zaXRvcnkzODE1NTE2OTA=\",\"name\":\"mattermost-plugin-github\",\"full_name\":\"sibasankarnayak/mattermost-plugin-github\",\"private\":false,\"owner\":{\"name\":\"sibasankarnayak\",\"email\":\"86650698+sibasankarnayak@users.noreply.github.com\",\"login\":\"sibasankarnayak\",\"id\":86650698,\"node_id\":\"MDQ6VXNlcjg2NjUwNjk4\",\"avatar_url\":\"https://avatars.githubusercontent.com/u/86650698?v=4\",\"gravatar_id\":\"\",\"url\":\"https://api.github.com/users/sibasankarnayak\",\"html_url\":\"https://github.com/sibasankarnayak\",\"followers_url\":\"https://api.github.com/users/sibasankarnayak/followers\",\"following_url\":\"https://api.github.com/users/sibasankarnayak/following{/other_user}\",\"gists_url\":\"https://api.github.com/users/sibasankarnayak/gists{/gist_id}\",\"starred_url\":\"https://api.github.com/users/sibasankarnayak/starred{/owner}{/repo}\",\"subscriptions_url\":\"https://api.github.com/users/sibasankarnayak/subscriptions\",\"organizations_url\":\"https://api.github.com/users/sibasankarnayak/orgs\",\"repos_url\":\"https://api.github.com/users/sibasankarnayak/repos\",\"events_url\":\"https://api.github.com/users/sibasankarnayak/events{/privacy}\",\"received_events_url\":\"https://api.github.com/users/sibasankarnayak/received_events\",\"type\":\"User\",\"site_admin\":false},\"html_url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github\",\"description\":\"GitHub plugin for Mattermost \",\"fork\":true,\"url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github\",\"forks_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/forks\",\"keys_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/keys{/key_id}\",\"collaborators_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/collaborators{/collaborator}\",\"teams_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/teams\",\"hooks_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/hooks\",\"issue_events_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/issues/events{/number}\",\"events_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/events\",\"assignees_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/assignees{/user}\",\"branches_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/branches{/branch}\",\"tags_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/tags\",\"blobs_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/git/blobs{/sha}\",\"git_tags_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/git/tags{/sha}\",\"git_refs_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/git/refs{/sha}\",\"trees_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/git/trees{/sha}\",\"statuses_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/statuses/{sha}\",\"languages_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/languages\",\"stargazers_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/stargazers\",\"contributors_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/contributors\",\"subscribers_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/subscribers\",\"subscription_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/subscription\",\"commits_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/commits{/sha}\",\"git_commits_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/git/commits{/sha}\",\"comments_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/comments{/number}\",\"issue_comment_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/issues/comments{/number}\",\"contents_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/contents/{+path}\",\"compare_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/compare/{base}...{head}\",\"merges_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/merges\",\"archive_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/{archive_format}{/ref}\",\"downloads_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/downloads\",\"issues_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/issues{/number}\",\"pulls_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/pulls{/number}\",\"milestones_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/milestones{/number}\",\"notifications_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/notifications{?since,all,participating}\",\"labels_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/labels{/name}\",\"releases_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/releases{/id}\",\"deployments_url\":\"https://api.github.com/repos/sibasankarnayak/mattermost-plugin-github/deployments\",\"created_at\":1625019880,\"updated_at\":\"2021-06-30T02:24:41Z\",\"pushed_at\":1633430998,\"git_url\":\"git://github.com/sibasankarnayak/mattermost-plugin-github.git\",\"ssh_url\":\"git@github.com:sibasankarnayak/mattermost-plugin-github.git\",\"clone_url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github.git\",\"svn_url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github\",\"homepage\":\"\",\"size\":2777,\"stargazers_count\":0,\"watchers_count\":0,\"language\":null,\"has_issues\":false,\"has_projects\":true,\"has_downloads\":true,\"has_wiki\":false,\"has_pages\":false,\"forks_count\":0,\"mirror_url\":null,\"archived\":false,\"disabled\":false,\"open_issues_count\":1,\"license\":{\"key\":\"apache-2.0\",\"name\":\"Apache License 2.0\",\"spdx_id\":\"Apache-2.0\",\"url\":\"https://api.github.com/licenses/apache-2.0\",\"node_id\":\"MDc6TGljZW5zZTI=\"},\"allow_forking\":true,\"visibility\":\"public\",\"forks\":0,\"open_issues\":1,\"watchers\":0,\"default_branch\":\"master\",\"stargazers\":0,\"master_branch\":\"master\"},\"pusher\":{\"name\":\"sibasankarnayak\",\"email\":\"86650698+sibasankarnayak@users.noreply.github.com\"},\"sender\":{\"login\":\"sibasankarnayak\",\"id\":86650698,\"node_id\":\"MDQ6VXNlcjg2NjUwNjk4\",\"avatar_url\":\"https://avatars.githubusercontent.com/u/86650698?v=4\",\"gravatar_id\":\"\",\"url\":\"https://api.github.com/users/sibasankarnayak\",\"html_url\":\"https://github.com/sibasankarnayak\",\"followers_url\":\"https://api.github.com/users/sibasankarnayak/followers\",\"following_url\":\"https://api.github.com/users/sibasankarnayak/following{/other_user}\",\"gists_url\":\"https://api.github.com/users/sibasankarnayak/gists{/gist_id}\",\"starred_url\":\"https://api.github.com/users/sibasankarnayak/starred{/owner}{/repo}\",\"subscriptions_url\":\"https://api.github.com/users/sibasankarnayak/subscriptions\",\"organizations_url\":\"https://api.github.com/users/sibasankarnayak/orgs\",\"repos_url\":\"https://api.github.com/users/sibasankarnayak/repos\",\"events_url\":\"https://api.github.com/users/sibasankarnayak/events{/privacy}\",\"received_events_url\":\"https://api.github.com/users/sibasankarnayak/received_events\",\"type\":\"User\",\"site_admin\":false},\"created\":true,\"deleted\":false,\"forced\":false,\"base_ref\":null,\"compare\":\"https://github.com/sibasankarnayak/mattermost-plugin-github/commit/3c642e9e366a\",\"commits\":[{\"id\":\"3c642e9e366a559f14374fa250ec8815151e7b0f\",\"tree_id\":\"2cbc8486830aa6555af2f2ce4be76fd681b263df\",\"distinct\":true,\"message\":\"Update README.md\",\"timestamp\":\"2021-10-05T16:19:58+05:30\",\"url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github/commit/3c642e9e366a559f14374fa250ec8815151e7b0f\",\"author\":{\"name\":\"sibasankarnayak\",\"email\":\"86650698+sibasankarnayak@users.noreply.github.com\",\"username\":\"sibasankarnayak\"},\"committer\":{\"name\":\"GitHub\",\"email\":\"noreply@github.com\",\"username\":\"web-flow\"},\"added\":[],\"removed\":[],\"modified\":[\"README.md\"]}],\"head_commit\":{\"id\":\"3c642e9e366a559f14374fa250ec8815151e7b0f\",\"tree_id\":\"2cbc8486830aa6555af2f2ce4be76fd681b263df\",\"distinct\":true,\"message\":\"Update README.md\",\"timestamp\":\"2021-10-05T16:19:58+05:30\",\"url\":\"https://github.com/sibasankarnayak/mattermost-plugin-github/commit/3c642e9e366a559f14374fa250ec8815151e7b0f\",\"author\":{\"name\":\"sibasankarnayak\",\"email\":\"86650698+sibasankarnayak@users.noreply.github.com\",\"username\":\"sibasankarnayak\"},\"committer\":{\"name\":\"GitHub\",\"email\":\"noreply@github.com\",\"username\":\"web-flow\"},\"added\":[],\"removed\":[],\"modified\":[\"README.md\"]}}"}

```